src/core/payoutCalculator/PayoutCalculatorIsolateSuperCharge.ts

```diff
@@ -105,7 +126,11 @@ export class PayoutCalculatorIsolateSuperCharge implements IPayoutCalculator {
                     const commissionRate = commissionRates[staker.publicKey]
                         ? commissionRates[staker.publicKey].commissionRate
                         : defaultCommissionRate;
-
+                    
+                    if (winner.publicKey == staker.publicKey && burnScr) { 
```

Comment:
A comment explaining the assumptions around why this is handled this way would be helpful. 
I think it is because we can assume that the winner will only be the staker once in this loop, so that is a good way to make sure we just burn once.
If there were ever a case where the winner could be iterated >1 time here, there would be a problem.